src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,856 @@
+                reportDiagnostic(diagnostic);
+            }
+
+            private static IEnumerable<INamedTypeSymbol> GetAttributesForMethodsToIgnore(Compilation compilation)
+            {
+                // Ignore conditional methods (One conditional will often call another conditional method as its only use of a parameter)
+                var conditionalAttribte = compilation.ConditionalAttribute();
+                if (conditionalAttribte != null)
+                {
+                    yield return conditionalAttribte;
+                }
+
+                // Ignore methods with special serialization attributes (All serialization methods need to take 'StreamingContext')
+                var onDeserializingAttribute = compilation.OnDeserializingAttribute();
+                if (onDeserializingAttribute != null)
+                {
+                    yield return onDeserializingAttribute;
+                }
+
+                var onDeserializedAttribute = compilation.OnDeserializedAttribute();
+                if (onDeserializedAttribute != null)
+                {
+                    yield return onDeserializedAttribute;
+                }
+
+                var onSerializingAttribute = compilation.OnSerializingAttribute();
+                if (onSerializingAttribute != null)
+                {
+                    yield return onSerializingAttribute;
+                }
+
+                var onSerializedAttribute = compilation.OnSerializedAttribute();
+                if (onSerializedAttribute != null)
+                {
+                    yield return onSerializedAttribute;
+                }
+
+                // Don't flag obsolete methods.
+                var obsoleteAttribute = compilation.ObsoleteAttribute();
+                if (obsoleteAttribute != null)
+                {
+                    yield return obsoleteAttribute;
+                }
+            }
+
+            private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
+            {
+                // Ignore implicitly declared parameters and methods, extern methods, abstract methods,
+                // virtual methods, overrides, interface implementations, accessors and lambdas.
+                if (parameter.IsImplicitlyDeclared ||
+                    parameter.Name == "_" ||
+                    !(parameter.ContainingSymbol is IMethodSymbol method) ||
+                    method.IsImplicitlyDeclared ||
+                    method.IsExtern ||
+                    method.IsAbstract ||
+                    method.IsVirtual ||
+                    method.IsOverride ||
+                    !method.ExplicitOrImplicitInterfaceImplementations().IsEmpty ||
+                    method.IsAccessor() ||
+                    method.IsAnonymousFunction())
+                {
+                    return false;
+                }
+
+                // Ignore event handler methods "Handler(object, MyEventArgs)"
```

Comment:
this is another 'what' comment.  it doesn't explain the 'why'.  For example, as the reader/maintainer/reviewer, i have no idea why we're ignoring these.